### ISSUE TITLE
Quick update workflow / .NET Core

### ISSUE THREAD

**Issue by MiloszKrajewski** [fsprojects/Paket#2880]
### Description

NOTE: this is about .NET Core. I believe, full framework doesn't have this problem, as framework assemblies are not queried.

I guess it's more like a question/feature request. Maybe there is already a way to do it I just didn't find it. 

I have 2+ repositories, let's say Framework and ProjectA (ProjectB, ProjectC, etc). ProjectA depends on Framework. They are all in rapid growth phase so they both get updated quite frequently, we would like, in ProjectA, to use new version on Framework as quickly as possible. 

We publish Framework to our local feed and update ProjectA with:

```paket update Framework.* --filter```

This works as expected it is just very slow (for now it takes ~8 minutes per project).

```
 - Resolver: 8 minutes, 25 seconds (1 runs)
    - Runtime: 6 minutes, 10 seconds
    - Blocked (retrieving package details): 2 minutes, 11 seconds (269 times)
    - Blocked (retrieving package versions): 4 seconds (5 times)
    - Not Blocked (retrieving package versions): 276 times
    - Not Blocked (retrieving package details): 12 times
 - Disk IO: 3 seconds
 - Average Request Time: 531 milliseconds
 - Number of Requests: 1074
 - Runtime: 8 minutes, 30 seconds
```

There are two sources of "slowness": *runtime* and *retrieving package details*.
I guess that *runtime* is actual resolution algorithm, while *retrieving package details* is contacting nuget and retrieving package metadata.

*NOTE:* It's actually quite interesting that *Runtime* takes that long.

My question/suggestion would be:
* option to update (--quick) only mentioned packages (Framework.*) without re-querying transitive dependencies as long as they didn't change
* caching package metadata responses from nuget (it does not change, right?)

I guess, to be sure it would require running "regular" update on the end of the day, but would speed up development in the meantime.

### Repro steps

1. Unzip attached repo
2. Run "setup.cmd" (it will build "framework" and "project")
3. Update "framework" in "project": ```cd pkt3p && paket update pkt* --filter```
4. NOTE: initial 'install' took less than 'update' (45s vs 2.5m on my machine)
5. NOTE: this time *runtime* took most of the time (2m)

NOTE: this is just ONE dependency.

### Known workarounds

Updating lock file manually? (cringe)

**Comment by MiloszKrajewski** [comment]
Forgot to attach project: [pkt3.zip](https://github.com/fsprojects/Paket/files/1434366/pkt3.zip)

**Comment by forki** [comment]
Cc @matthid

Am 01.11.2017 15:05 schrieb "Milosz Krajewski" <notifications@github.com>:

**Comment by MiloszKrajewski** [comment]
Some observation: it is get very slow when updating subset of packages, for example (using attached project):
* ```paket update```: 45s
* ```paket update x*x --filter```: 2.5m (NOTE: none of the packages match 'x*x') - with CPU utilization 100% (single core)

I've also spotted #2858 with similar issue, so I think there is definitely a problem.

(Caching packages metadata is unrelated, although might be useful anyway)

**Comment by forki** [comment]
What makes you think we don't cache already?

Am 02.11.2017 00:25 schrieb "Milosz Krajewski" <notifications@github.com>:

**Comment by MiloszKrajewski** [comment]
I giess this: ```Blocked (retrieving package details): 2 minutes, 11 seconds (269 times)```. No versions has been changed, so I assumed it does exactly the same nuget API calls. I'm not talking about caching .nupkg, but metadata so I could imagine (almost?) no network activity for this operation.

**Comment by forki** [comment]
We do cache meta data. But you can't cache questions like "what versions
are available" right now.

Am 02.11.2017 10:45 schrieb "Milosz Krajewski" <notifications@github.com>:

**Comment by matthid** [comment]
> No versions has been changed

Yes how would we know that? We know after we asked nuget for latest versions.

however, yes the packages details should be cached and there should be no reason for paket to block. this only is reasonable if paket encountered a lot of new versions...

**Comment by matthid** [comment]
regarding the partial update, yes we should investigate. we already have some tests in our Integration test suite which have become quite slow

**Comment by MiloszKrajewski** [comment]
@matthid 

1. You are absolutely right ("Yes how would we know that?"). I get that. What I was suggesting was a potential new feature/option for ```paket update``` which tell paket to assume that there are no new versions as long as old ones satisfy criteria.

2. There is something weird about ```paket update <patter> --filter```. It is much slower and spends much more time in "runtime" than regular ```paket update```.

```
./paket update x*x --filter
...
Runtime: 2 minutes, 9 seconds
```

vs

```
./paket update
...
Runtime: 2 seconds
```

NOTE: 2 seconds vs 2 minutes. 

3. I assumed (wrongly) that package details are not cached. 
I can see them now in ```C:\Users\<user>\AppData\Local\NuGet\Cache```.

I did that because it takes suspiciously long time even if running ```paket update``` twice in a row. 

first run:
```
./paket update
...
Blocked (retrieving package details): 28 seconds (68 times)
Blocked (retrieving package versions): 3 seconds (7 times)
```

second run (1s second later):
```
./paket update
...
Blocked (retrieving package details): 26 seconds (62 times)
Blocked (retrieving package versions): 4 seconds (8 times)
````

third run (1s second later):
```
./paket update
...
Blocked (retrieving package details): 28 seconds (68 times)
Blocked (retrieving package versions): 2 seconds (6 times)
```

It seems like it keeps retrieving package details.
